src/urllib3/_collections.py

```diff
@@ -155,7 +155,7 @@ def __setitem__(self, key, val):
 
     def __getitem__(self, key):
         val = self._container[key.lower()]
-        return ", ".join([six.ensure_str(v, "ascii") for v in val[1:]])
+        return ", ".join(val[1:])
```

Comment:
So this will still explode if passed bytes, which is fine (???) because it's the same behavior as in 1.25.x and before. cc @jalopezsilva

Reply:
All checks are now using ``getall`` which works around the problem that I saw. We won't see ``_getitem__`` throw if the User-Agent is bytes. 

That works but it feels fragile. Shouldn't we respect whatever type the user provided? If bytes were used, let's use bytes, otherwise use strings. 

You can do something like to keep to avoid throwing if bytes.
```
b", ".join
```

Reply:
Hmm, yeah we could maybe do something there? Basically follow this logic:

- Get the list of headers
- If they're all bytes return `b", ".join(val[1:])`
- Otherwise return `", ".join(val[1:])`

This way if a user uses mixed types they still get the same explosion? I'd like a little poking around with what happens when httplib is passed both bytes and str headers on Python 2.x since it had strange behavior when doing so with HTTP method and URL. Probably deserves it's own issue!

Reply:
I captured this issue here: https://github.com/urllib3/urllib3/issues/2021